effects/stage_effect.py

```diff
@@ -0,0 +1,55 @@
+from game import Game
+from effects.effect import Effect
+from citizens.citizen import Citizen
+from citizens.player import Player
+import user_interaction
+import utils
+
+
+class StageEffect(Effect):
+    def __init__(self, name: str, game: Game, creator: Citizen):
+        Effect.__init__(self, name, game, creator)
+
+    def activate(self, target_number):
+        target_number = user_interaction.read_index('config.type_target')
+        while(not self.validate(target_number)):
+            user_interaction.show_active_instant("config.type_error")
+            target_number = user_interaction.read_index("config.type_target")
+
+        self.target = self.game.citizens[target_number - 1]
+
+    def resolve(self):
+        self.creator._staging_available = False
+        # self.creator.allowed_actions.remove(CARD_USAGE)
+
+        if isinstance(self.target, Player):
+            self.target.hp -= 1
+            user_interaction.save_passive('config.lose_hp')
+            user_interaction.save_global('config.failed_stage')
+
+        else:
+            self.creator.name, self.target.name = self.target.name, self.creator.name
+            self.target.hp -= 1
+            if self.creator.citizen_card is not None:
+                self.creator.stolen_cards.append(self.creator.citizen_card)
+                self.creator.citizen_card = None
+
+            self.creator.citizen_card = self.target.citizen_card
+
+            creator_index = [index for index, citizen in enumerate(
+                self.game.citizens) if citizen is self.creator][0]
+            target_index = [index for index, citizen in enumerate(
+                self.game.citizens) if citizen is self.target][0]
```

Comment:
We should move this logic to utils.
Something like
utils.find_citizen_index(target_citizen: Citizen, citizens: List[Citizen])

Reply:
Done